src/oop/oop2.py

```diff
@@ -4,12 +4,13 @@
 # object is constructed.
 
 class GroundVehicle():
-    def __init__(self, num_wheels):
-        self.num_wheels = num_wheels
+    def __init__(self, num_wheels = None):
+        self.num_wheels = num_wheels if num_wheels is not None else 4
```

Comment:
Nice! This absolutely works, however I think a more readable solution is to simply put the 4 as the default argument for the `num_wheels` parameter. Well done using python's ternary conditional expression here.